### PR TITLE
[CF-430] Improve configuration validation

### PR DESCRIPTION
**Pull request by antonf**
This patch validate configuration file more strictly, not passing
values that are not specified in schema. Also for fields that
specify class names the classes are loaded early in order to
detect error loading module/class during field validation and not
during object initialization.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferry.lib.utils.bases
cloudferry/lib/utils/bases.py:14: [W0611(unused-import), ] Unused import collections
************* Module cloudferry.config
cloudferry/config.py:57: [W0613(unused-argument), ConfigSchema.check_unknown_fields] 
Unused argument 'data'
cloudferry/config.py:69: [E1102(not-callable), ConfigSchema.to_config_section] 
self.config_section_class is not callable
cloudferry/config.py:142: [E1102(not-callable), ConfigSection.__init__] 
self.schema_class is not callable
cloudferry/config.py:256: [E1101(no-member), 
Scope.check_migration_have_correct_source_and_dict] Instance of 'Scope' has no 'fields' 
member
cloudferry/config.py:258: [E1101(no-member), 
Scope.check_migration_have_correct_source_and_dict] Instance of 'Scope' has no 'fields' 
member
cloudferry/config.py:378: [E1102(not-callable), load] Configuration.schema_class is not 
callable

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3478/

File: cloudferry/config.py [hunk 1]
```diff
@@ -38,6 +39,134 @@
...
 
 
+class ValidationError(Exception):
+    def __str__(self):
+        try:
+            if self.args:
+                return self.args[0] % self.args[1:]
+            else:
+                return 'ValidationError'
+        except Exception:  # pylint: disable=broad-except
```
Comment (roman-verchikov) [inline cloudferry/config.py]: I'm suspect catching `TypeError` here should be enough, `Exception` is way too broad

**Comment by roman-verchikov** [comment]
Would be good to have unit tests which cover introduced functionality

**Comment by roman-verchikov** [comment]
Please resolve inline and pep8

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by MirantisWorkloadMobility** [comment]
$PULL_STATUS
$BUILD_URL

**Comment by MirantisWorkloadMobility** [comment]
$PYLINT_STATUS

**Comment by antonf** [comment]
Functional tests passed: http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-icehouse-to-juno-migrate2/39/

**Comment by antonf** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3629/

**Comment by roman-verchikov** [comment]
retest

**Comment by MirantisWorkloadMobility** [comment]
Merge check failed
http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-unittests-and-style-checks/3647/

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by antonf** [comment]
Tests successful: http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-grizzly-to-juno-migrate2/53/ http://jenkins-cf.vm.mirantis.net:8080/job/cloudferry-functional-tests-icehouse-to-juno-migrate2/11/

**Comment by roman-verchikov** [comment]
+1

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**